server.go

```diff
@@ -138,23 +142,25 @@ func NewFromConfig(conf Config) (ret Server, err error) {
 		})
 
 		if err != nil {
-			ret.logger.Info("error getting AWS session: %s", err)
+			log.Info("error getting AWS session: %s", err)
 			svc = nil
 		} else {
-			ret.logger.Info("Successfully created AWS session")
+			log.Info("Successfully created AWS session")
 			svc = s3.New(sess)
 
 			// TODO(aditya) store this on the server
 			S3Bucket = conf.AWSBucket
 		}
 	} else {
-		ret.logger.Info("AWS credentials not found")
+		log.Info("AWS credentials not found")
 	}
 
 	if svc == nil {
-		ret.logger.Info("S3 archives are disabled")
+		plugin := S3Plugin{svc: svc}
+		plugin.Initialize(ret.statsd, log)
```

Comment:
Originally I had made this a requisite method for the interface - it's a way of indicating that types which satisfy an interface are intended to have certain fields (since you can't require fields on plugins). 

This way, we could do, e.g.

```go
for _, plugin := range []plugin{&S3Plugin{svc}, &StatelessPlugin{}}{
    plugin.Initialize(statsd, log}
}
```


But it's overkill, so I'll remove it.